### PR TITLE
[BBC-11165] Add `ColorIcon`

### PR DESCRIPTION
**Pull request by mcellier**
## Description

Capture d’écran 2020-11-20 à 11 44 48

## What has been done

New icon using `iconColor` to fill the icon. Added a new `borderColor` prop to this icon to handle some specific cases

## Things to consider

The white is the only color that uses a border (black). On iOS, the border is set on the black icon when the user is in dark mode. I chose to stay generic in the library.

## How it was tested

Storybook

### CODE REVIEW

File: src/icon/colorIcon.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    borderColor?: string
+  }>
+
+export const ColorIcon = (props: ColorIconProps) => (
+  <BaseIcon {...props}>
+    <circle
+      cx="12"
+      cy="12"
+      r="9.5"
+      stroke={props.borderColor || props.iconColor}
```
Comment (gabrielagabriel) [inline src/icon/colorIcon.tsx:16]: please use **CSS**(in-js) for that

File: src/icon/colorIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+
+import { BaseIcon, BaseIconDefaultProps, Icon } from '../_utils/icon'
+
+export type ColorIconProps = Icon &
+  Readonly<{
+    borderColor?: string
```
Comment (gabrielagabriel) [inline src/icon/colorIcon.tsx:7]: Also which colors are allowed? the whole palette? Should you type the colors?

**Comment by wakooka** [comment]
Do you think we can use `CircleIcon` instead with a few modifications?

**Comment by mcellier** [comment]
> Do you think we can use `CircleIcon` instead with a few modifications?

I thought about it, but `CircleIcon` already has LOT of logic in it (spinning & so on). I prefered the smaller, more specialized approach. Especially if one day we want to be smarter about the border thing (dark mode)

File: src/icon/colorIcon.tsx (hunk 2 above)
Comment (mcellier) [reply src/icon/colorIcon.tsx:7]: Color is sent by the backend (as explained in the FADR). We don't know what they'll send us, even if I don't think it changes that often.

File: src/icon/colorIcon.tsx (hunk 1 above)
Comment (mcellier) [reply src/icon/colorIcon.tsx:16]: Same answer as my previous PR.

**Review by wakooka — APPROVED** [review]

**Review by PerrineBoissieres — APPROVED** [review]

**Review by Tuxosaurus — APPROVED** [review]

File: src/icon/colorIcon.tsx (hunk 1 above)
Comment (gabrielagabriel) [reply src/icon/colorIcon.tsx:16]: not the same issue! here is even more problematic since we are treating fill and stroke

**Pull request merged**